sqlcollaborative/dbatools
# DBAServicePrincipalName

Fixes # 625

Changes proposed in this pull request:
 - Adds new cmdlets for SPNs!
 - Get-DBAServicePrincipalName - Returns found SPNs for given computer/account
 - Test-DBAServicePrincipalName - Returns "required" SPNS for a given computer (and all instances found on it)
- Set-DBAServicePrincipalName - Sets SPN for given AD account and also enables delegation to said spns.

How to test this code: 
- Run the Get- command to see if SPNs already exist, and validate against AD
- Run the Test- to see what SPNs it thinks it should have, and for IsSet "true" and "false" validate in AD
- Run the Set- to add SPNs and delegation to test accounts

Has been tested on minimum requirements:
- [ ]  Powershell 3
- [ ]  Windows 7
- [ ]  SQL Server 2000

Has been tested on maximum requirements:
- [ ]  SQL Server vNext
- [x ]  Windows 10
- [ ]  Azure Database

Tests for tester:
